src/query/functions/src/scalars/boolean.rs

```diff
@@ -97,6 +97,17 @@ pub fn register(registry: &mut FunctionRegistry) {
         "and",
         FunctionProperty::default(),
         |lhs, rhs| {
+            let all_false =  lhs.value.as_ref().map(|v| v.has_false && !v.has_true).unwrap_or(false)
```

Comment:
It could be simplify to:

```rust
let lhs_has_null = lhs.has_null;
let lhs_has_true = lhs.value.as_ref().map(|v| v.has_true).unwrap_or(false);
let lhs_has_false = lhs.value.as_ref().map(|v| v.has_false).unwrap_or(false);

let rhs_has_null = rhs.has_null;
let rhs_has_true = rhs.value.as_ref().map(|v| v.has_true).unwrap_or(false);
let rhs_has_false = rhs.value.as_ref().map(|v| v.has_false).unwrap_or(false);

let has_null = lhs_has_null || rhs.has_null;
let has_true = lhs_has_true && rhs_has_true;
let has_false = lhs_has_false || rhs_has_false;

let value = if has_true || has_false {
    Some(Box::new(BooleanDomain{
        has_true,
        has_false,
    }))
} else {
    None
};

NullalbeDomain {
    has_null,
    value,
}
```

Reply:
`has_null` should be false if `has_false` and `!has_true`

Reply:
updated